internal/pkg/errors/template_errors.go

```diff
@@ -0,0 +1,215 @@
+package errors
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+	"text/template"
+
+	"github.com/hashicorp/nomad-pack/internal/pkg/variable/parser"
+)
+
+type PackTemplateError struct {
+	Filename    string   // template filename returned
+	Line        int      // the line in the template if found
+	StartChar   int      // the character number in the template if found
+	EndChar     int      // the character number calculated as the end if an "at" is found
+	Err         error    // the last element in given error text when split by ": "
+	Details     string   // some additional help text for specific known error patterns
+	Suggestions []string // some suggestions to add to the error context
+	Extra       []string // remaining splits between the beginning elements and the last one which is the error
+
+	origErr    error
+	badElement string
+	at         string
+	tplctx     parser.PackTemplateContext
+}
+
+func (p PackTemplateError) Error() string {
+	return p.Err.Error()
+}
+
+func ParseTemplateError(tplCtx parser.PackTemplateContext, err error) PackTemplateError {
+	out := PackTemplateError{Err: err, origErr: err, tplctx: tplCtx}
+
+	var execErr template.ExecError
+	if errors.As(err, &execErr) {
+		out.parseExecError(execErr)
+	}
+
+	return out
+}
+
+func (p PackTemplateError) ToWrappedUIContext() *WrappedUIContext {
```

Comment:
vide James' comment above: make methods of `PackTemplateError` consistent, i.e., they should all be pointer receivers.